src/compiler/interpreter.jl

```diff
@@ -214,21 +214,27 @@ function Core.Compiler.abstract_call_gf_by_type(
     callinfo = ret.info
     method_table = Core.Compiler.method_table(interp)
     specTypes = simplify_kw(atype)
+    caller = if callinfo isa Core.Compiler.MethodMatchInfo && callinfo.results isa Core.Compiler.MethodLookupResult
```

Comment:
This seems wrong. We want to recompile the method one-above in the call-graph not the primal method from whence we came?

Reply:
I don’t follow? If a rule is defined for f we want in invalidation edge exist from inactive(f) -> f, right?

Reply:
For example if f defines a custom rule, we need to invalidate autodiff(f) to now just call that rule